lib/question.js

```diff
@@ -1522,7 +1532,7 @@ module.exports = {
             const locals = {};
             config.setLocals(locals);
             _.assign(locals, this._buildQuestionUrls(urlPrefix, variant, question, instance_question, assessment));
-            _.assign(locals, this._buildLocals(variant, question, instance_question, assessment, assessment_instance, assessment_question));
+            _.assign(locals, this._buildLocals(variant, question, instance_question, assessment, assessment_instance, assessment_question, null));
```

Comment:
I'm not sure if `null` is appropriate here... I may be a bit paranoid, but I imagine if a grading job that takes a really long time could cause a page that was supposed to be non-submittable to end up showing a submit button.

Reply:
I can't think of a better way to handle this. I guess `null` is an appropriate heuristic in this case, as the corner cases where this happens to affect the result can then be caught in submission errors instead.